### DISCUSSION TITLE
"YOU'VE EXCEEDED THE CHAT LIMIT" -- Cannot utilize the context window....

### DISCUSSION THREAD

**Discussion by zdondada** [community/community#138352] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Question

### Body

I was trying to paste my book into the interface (it's under 120k tokens) and it won't let me. So, I tried half of it, and it still won't let me. There's no point in having it there if we can't use it...

![image](https://github.com/user-attachments/assets/6ac9891f-f16a-475f-8c13-475869ff3a53)

**Comment by matthewisabel** [comment]
Sorry for the confusion here. We'll make an update to provide clearer errors!

While the model supports a larger token limit than what is currently constrained, we have a lower threshold currently: https://docs.github.com/en/github-models/prototyping-with-ai-models#rate-limits

We're still evaluating these limits as this is limited preview and there isn't a seamless upgrade path yet to let you go beyond these constraints. Hopefully more updates soon and I'll think about how we can provide more clarity here.

**Reply by amy-why-msft** [reply]
At least display what's the limit in the UI so we can see and adjust accordingly. I had to use trial and error to find the limit, for o1 it seems it is ~3000 tokens. 4o is ~8000 tokens?
